flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/models/NotificationChannelPlugin.java

```diff
@@ -0,0 +1,89 @@
+package com.dexterous.flutterlocalnotifications.models;
+
+import android.graphics.Color;
+
+import com.dexterous.flutterlocalnotifications.SoundSource;
+
+import java.util.Map;
+
+public class NotificationChannelPlugin {
+    private static final String CHANNEL_ID = "channelId";
+    private static final String CHANNEL_NAME = "channelName";
+    private static final String CHANNEL_DESCRIPTION = "channelDescription";
+    private static final String CHANNEL_SHOW_BADGE = "channelShowBadge";
+    private static final String IMPORTANCE = "importance";
+    private static final String PLAY_SOUND = "playSound";
+    private static final String SOUND = "sound";
+    private static final String SOUND_SOURCE = "soundSource";
+    private static final String ENABLE_VIBRATION = "enableVibration";
+    private static final String VIBRATION_PATTERN = "vibrationPattern";
+    private static final String CHANNEL_ACTION = "channelAction";
+    private static final String ENABLE_LIGHTS = "enableLights";
+    private static final String LED_COLOR_ALPHA = "ledColorAlpha";
+    private static final String LED_COLOR_RED = "ledColorRed";
+    private static final String LED_COLOR_GREEN = "ledColorGreen";
+    private static final String LED_COLOR_BLUE = "ledColorBlue";
+
+    public String channelId = "Default_Channel_Id";
```

Comment:
I made some comments on the Flutter side (as that's what I looked at first) that also apply here where the `channel` prefix for the name of the fields aren't necessary. `channelAction` can be kept though and set to `NotificationChannelAction.CreateIfNotExists` when `NotificationChannelPlugin from(Map<String, Object> ...)` is called since this class is used when a channel is creating just before a notification is shown